actionpack/lib/action_dispatch/testing/assertions/response.rb

```diff
@@ -85,17 +86,13 @@ def normalize_argument_to_redirection(fragment)
           end
         end
 
-        def generate_response_message(type)
-          message = "Expected response to be a <#{type}>, but was"
-
+        def response_message
           if @response.redirection?
             redirect_is = normalize_argument_to_redirection(@response.location)
-            message << " a redirect to <#{redirect_is}>"
+            "<302> redirected to <#{redirect_is}>"
```

Comment:
@matthewd @jonatack Thanks for the followup. You are right. 

I think we will have to check if `@response.response_code` matches with `302` or `Rack::Utils::SYMBOL_TO_STATUS_CODE[:found]` instead of checking for `redirection?`. 

But it will become too specific and increase code complexity. We should possibly revert the change   as it's too specific for a particular case.